Allure.XUnit/AllureXunitFacade.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+using Xunit.Abstractions;
+
+#nullable enable
+
+namespace Allure.XUnit
+{
+    public static class AllureXunitFacade
+    {
+        public static IMessageSink CreateAllureXunitMessageHandler(
+            IRunnerLogger logger
+        )
+        {
+            AllureXunitPatcher.PatchXunit(logger);
+            var secondReporter = ResolveSecondReporter();
+            var (startupMessage, sink) = ResolveMessageAndSink(
+                new AllureMessageSink(logger),
+                secondReporter,
+                logger
+            );
+            logger.LogImportantMessage(startupMessage);
+            return sink;
+        }
+
+        static IRunnerReporter? ResolveSecondReporter() =>
+            AllureXunitConfiguration.CurrentConfig.XunitRunnerReporter switch
+            {
+                "none" => null,
+                "auto" => ResolveAutoReporter(),
+                string reporterName => ResolveExplicitReporter(reporterName)
+            };
+
+        static (string, IMessageSink) ResolveMessageAndSink(
+            AllureMessageSink allureSink,
+            IRunnerReporter? secondReporter,
+            IRunnerLogger logger
+        ) =>
+            secondReporter is null ? (
+                ALLURE_REPORTER_ON_MSG,
+                allureSink
+            ) : (
+                $"{ALLURE_REPORTER_ON_MSG}. " +
+                    $"Secondary reporter: {secondReporter.RunnerSwitch}",
+                new ComposedMessageSink(
+                    allureSink,
+                    secondReporter.CreateMessageHandler(logger)
+                )
+            );
+
+        static IRunnerReporter? ResolveAutoReporter() =>
+            (
+                from reporter in GetReporters()
+                where reporter.IsEnvironmentallyEnabled
+                select reporter
+            ).FirstOrDefault();
+
+        static IRunnerReporter ResolveExplicitReporter(string reporterName) =>
+            TryCreateReporterByType(
```

Comment:
Too complicated to read; Please, don't use several "??" operators in one call;
It's better to make 3 steps:
- try to create reporter by type
- try to create reporter by name
- throw exception if no reporters was created

Reply:
Yep, agree. At least with complex sub-expressions and misalignment that should be avoided.
Should be more readable now. Check it out, please:

https://github.com/allure-framework/allure-csharp/pull/382/files#diff-25f946e3f601f2b1ed798bf9cca8714e89aa513511c138ee56d62a22d33313d6R61-R73